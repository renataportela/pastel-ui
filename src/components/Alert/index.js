export { default, ALERT_COLORS } from './Alert'