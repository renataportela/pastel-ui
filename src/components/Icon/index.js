export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';