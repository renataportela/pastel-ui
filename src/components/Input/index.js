export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
