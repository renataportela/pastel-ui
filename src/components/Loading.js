import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading