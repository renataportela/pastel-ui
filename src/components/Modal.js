import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;