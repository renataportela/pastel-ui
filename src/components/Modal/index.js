export { default } from './Modal'
export { default as ModalBase } from './ModalBase'