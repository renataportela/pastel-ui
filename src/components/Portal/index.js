export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'