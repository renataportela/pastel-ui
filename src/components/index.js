// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'