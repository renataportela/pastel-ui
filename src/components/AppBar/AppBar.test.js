import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, AppBar, Menu } from '~/components'

describe('AppBar component', () => {
  it('renders an AppBar with brand Brand, text "left" and a Login button on the right', () => {
    const tree = renderer.create(
      <Theme>
        <AppBar 
          brand={<span>Brand</span>}
          onToggleMenu={() => {}}
          left="left"
          right={(
            <AppBar.Button label="Login" />
          )}
        />
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an AppBar with brand Brand, text "left" and a menu on the right', () => {
    const tree = renderer.create(
      <Theme>
        <AppBar 
          brand={<span>Brand</span>}
          onToggleMenu={() => {}}
          color="secondary"
          left={<div>left</div>}
          right={
            <Menu activator={<AppBar.Button label="Options" />}>
              <Menu.Item onClick={() => {}}>Option 1</Menu.Item>
              <Menu.Item onClick={() => {}}>Option 2</Menu.Item>
              <Menu.Item onClick={() => {}}>Option 3</Menu.Item>
            </Menu>
          }
        />
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})