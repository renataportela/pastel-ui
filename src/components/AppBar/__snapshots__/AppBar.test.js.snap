// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component renders an AppBar with brand Brand, text "left" and a Login button on the right 1`] = `
<div
  className="BaseColors__BaseStyle-s1fbie-0 cKbdWp AppBar__ColorBar-sc-1dfcr7j-0 cHsMAO"
>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__MenuIconSlot-sc-1dfcr7j-5 hJFspI"
  >
    <button
      className="ButtonStyle-sc-120zjl4-0 igBNNn"
      color="primary"
      disabled={false}
      kind="ghost"
      onClick={[Function]}
      size="lg"
    >
      <svg
        className="Svg__SvgStyle-pyey0i-0 gYARyh"
        fill="none"
        height="24px"
        size={null}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
      <span
        className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
        onMouseDown={[Function]}
      />
    </button>
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__BrandSlot-sc-1dfcr7j-4 jYGbRM"
  >
    <span>
      Brand
    </span>
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__LeftSlot-sc-1dfcr7j-2 jDKneH"
  >
    left
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__RightSlot-sc-1dfcr7j-3 jAbhgh"
  >
    <button
      className="ButtonStyle-sc-120zjl4-0 bMxttm AppBarButton-sc-642fva-0 dFEtZr"
      color="primary"
      disabled={false}
      kind="ghost"
      size="md"
    >
      Login
      <span
        className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
        onMouseDown={[Function]}
      />
    </button>
  </div>
</div>
`;

exports[`AppBar component renders an AppBar with brand Brand, text "left" and a menu on the right 1`] = `
<div
  className="BaseColors__BaseStyle-s1fbie-0 dqdQsy AppBar__ColorBar-sc-1dfcr7j-0 cHsMAO"
>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__MenuIconSlot-sc-1dfcr7j-5 hJFspI"
  >
    <button
      className="ButtonStyle-sc-120zjl4-0 igBNNn"
      color="secondary"
      disabled={false}
      kind="ghost"
      onClick={[Function]}
      size="lg"
    >
      <svg
        className="Svg__SvgStyle-pyey0i-0 gYARyh"
        fill="none"
        height="24px"
        size={null}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
      <span
        className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
        onMouseDown={[Function]}
      />
    </button>
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__BrandSlot-sc-1dfcr7j-4 jYGbRM"
  >
    <span>
      Brand
    </span>
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__LeftSlot-sc-1dfcr7j-2 jDKneH"
  >
    <div>
      left
    </div>
  </div>
  <div
    className="AppBar__Slot-sc-1dfcr7j-1 AppBar__RightSlot-sc-1dfcr7j-3 jAbhgh"
  >
    <button
      className="ButtonStyle-sc-120zjl4-0 bMxttm AppBarButton-sc-642fva-0 dFEtZr"
      color="primary"
      disabled={false}
      kind="ghost"
      onClick={[Function]}
      size="md"
    >
      Options
      <span
        className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
        onMouseDown={[Function]}
      />
    </button>
  </div>
</div>
`;
